tests/apps/test_similarity.py

```diff
@@ -222,217 +223,474 @@ def test_event_cardinality(photons, max_count, modes, expected):
+
+        assert not all([isinstance(op, sf.ops.LossChannel) for op in p_func.circuit])
+
+    def test_max_loss(self):
+        """Test if function samples from the vacuum when maximum loss is applied."""
+        dim = 5
+        graph = nx.complete_graph(dim)
+        state = similarity._get_state(graph, 0, 1)
+        cov = state.cov()
+        disp = state.displacement()
+
+        assert np.allclose(cov, 0.5 * state.hbar * np.eye(2 * dim))
+        assert np.allclose(disp, np.zeros(dim))
+
+
+class TestProbOrbitExact:
+    """Tests for the function ``strawberryfields.apps.similarity.prob_orbit_exact``"""
+
+    def test_invalid_n_mean(self):
+        """Test if function raises a ``ValueError`` when the mean photon number is specified to
+        be negative."""
+        g = nx.complete_graph(5)
+        with pytest.raises(ValueError, match="Mean photon number must be non-negative"):
+            similarity.prob_orbit_exact(g, [1, 1, 1, 1], n_mean=-1)
+
+    def test_invalid_loss(self):
+        """Test if function raises a ``ValueError`` when the loss parameter is specified outside
+        of range."""
+        g = nx.complete_graph(5)
+        with pytest.raises(ValueError, match="Loss parameter must take a value between zero and one"):
+            similarity.prob_orbit_exact(g, [1, 1, 1, 1], loss=2)
+
+    def test_prob_vacuum_orbit(self):
+        """Tests if the function gives the right probability for the empty orbit when the GBS
+        device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(5)
+        assert similarity.prob_orbit_exact(graph, [], 0) == 1.0
+
+    @pytest.mark.parametrize("k", [1, 3, 5, 7, 9])
+    def test_odd_photon_numbers(self, k):
+        """Test if function returns zero probability for odd number of total photons."""
+        graph = nx.complete_graph(10)
+        assert similarity.prob_orbit_exact(graph, [k]) == 0.0
+
+    def test_correct_result_returned(self, monkeypatch):
+        """Tests if the call to _get_state function is performed correctly and probabilities over all
+        permutations of the given orbit are summed correctly. The test monkeypatches the fock_prob
+        function so that the probability is the same for each sample permutation and
+        is equal to 1/8. For a 4-mode graph, [1, 1] has 6 possible permutations. """
+        graph = nx.complete_graph(4)
+        with monkeypatch.context() as m:
+            m.setattr("strawberryfields.backends.gaussianbackend.states.GaussianState.fock_prob", lambda *args, **kwargs: 1/8)
+            assert similarity.prob_orbit_exact(graph, [1, 1]) == 6/8
+
+    def test_known_result(self):
+        """Tests if the probability of detecting photons in nearly all modes of a
+        graph with a small mean photon number is correctly reproduced as
+        almost zero."""
+        graph = nx.complete_graph(4)
+        assert np.allclose(similarity.prob_orbit_exact(graph, [4], 1), 0)
```

Comment:
We could probably raise even higher than 4. But since the test passes I don't mind.